scmrepo/git/backend/pygit2.py

```diff
@@ -465,16 +465,25 @@ def _stash_push(
             self.repo.stash_drop()
         return str(oid), False
 
-    def _stash_apply(self, rev: str):
-        from pygit2 import GitError
+    def _stash_apply(
+        self,
+        rev: str,
+        reinstate_index: bool = False,
+        skip_conflicts: bool = False,
+        **kwargs,
+    ):
+        from pygit2 import GIT_CHECKOUT_ALLOW_CONFLICTS, GitError
 
         from scmrepo.git import Stash
 
         def _apply(index):
             try:
                 self.repo.index.read(False)
+                strategy = self._get_checkout_strategy()
+                if skip_conflicts:
+                    strategy |= GIT_CHECKOUT_ALLOW_CONFLICTS
```

Comment:
Note that this is pygit specific right now because there is no CGit `git stash apply` flag that gives this equivalent behavior.

libgit2 provides it since it's useful for programmatic reasons (where you need to automate everything vs being able to do interactive 3-way merge conflict resolutions with CLI git)